core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -235,6 +254,48 @@ public Result visit(Join e) {
     return result(join, leftResult, rightResult);
   }
 
+  private boolean isUsingOperator(Join e) {
+    return RexCall.class.isInstance(e.getCondition())
+        && ((RexCall)e.getCondition()).getOperator() == SqlLibraryOperators.USING;
+  }
+
+  private boolean checkForValidUsingOperands(RexNode condition, Context leftContext,
+      Context rightContext,  Map<SqlNode, SqlNode> usingSourceTargetMap) {
+    List<RexNode> usingOperands = ((RexCall)condition).getOperands();
```

Comment:
as discussed we should use this logic in the RelShuttle of Raven-Sql after the Equivalence service has done it's job.

Reply:
As discussed moving this to relshuttle would be difficult as metadata is not present and there would be a clash while comparing for same column in for two join tables.